compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1082,21 +1082,36 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         let hir = self.tcx.hir();
         let parent_node = hir.get_parent_node(obligation.cause.body_id);
         let node = hir.find(parent_node);
+        // FIXME(estebank): When encountering a method with a trait
+        // bound not satisfied in the return type with a body that has
+        // no return, suggest removal of semicolon on last statement.
+        // Once that is added, close #54771.
```

Comment:
I think the comment can now be removed, as the suggestion properly accounts for the type of the tail expression matching the trait now.